panic in go/types.(*Package).Path(...) with a well-formed source file and rule

Here's a minimized source file:
```
package store
import (
	"io"
	"io/ioutil"
	"strings"
	"testing"
)
type Store interface {
	Create(key string, expectedChecksum string, r io.ReadCloser) error
}
func content(s string) io.ReadCloser {
	return ioutil.NopCloser(strings.NewReader(s))
}
func TestStore(t *testing.T, s Store, kind string) {
	if err := s.Create("c", "deadbeef", content("3")); err == nil {
		t.Errorf("store %v: nil error from invalid checkum", kind)
	}
}
```

And here is a check with panic output:
```
~/go/src/github.com/Quasilyte/go-ruleguard/cmd/ruleguard/store $ ../ruleguard  -e 'm.Match("$t0 == $t1").Where(m["t0"].Type.Is("time.Time")).Report("using == with time.Time")'  .
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12d7f84]

goroutine 257 [running]:
go/types.(*Package).Path(...)
	/Users/dgryski/go/src/go.googlesource.com/go/src/go/types/package.go:30
github.com/quasilyte/go-ruleguard/ruleguard/typematch.(*Pattern).matchIdentical(0xc0002f4600, 0xc00042b0e0, 0x1471f20, 0xc00005abe0, 0xc00005abe0)
	/Users/dgryski/go/src/github.com/quasilyte/go-ruleguard/ruleguard/typematch/typematch.go:328 +0x824
github.com/quasilyte/go-ruleguard/ruleguard/typematch.(*Pattern).MatchIdentical(0xc0002f4600, 0x1471f20, 0xc00005abe0, 0x0)
	/Users/dgryski/go/src/github.com/quasilyte/go-ruleguard/ruleguard/typematch/typematch.go:234 +0x71
github.com/quasilyte/go-ruleguard/ruleguard.(*rulesParser).walkFilter.func6(0x1471f20, 0xc00005abe0, 0xc00042b380, 0x1471f20)
	/Users/dgryski/go/src/github.com/quasilyte/go-ruleguard/ruleguard/parser.go:505 +0x46
github.com/quasilyte/go-ruleguard/ruleguard.(*rulesRunner).handleMatch(0xc0000105c0, 0x0, 0x0, 0xc0002f4800, 0xc00049cce1, 0x17, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/dgryski/go/src/github.com/quasilyte/go-ruleguard/ruleguard/runner.go:103 +0x874
github.com/quasilyte/go-ruleguard/ruleguard.(*rulesRunner).run.func2.1(0x1471160, 0xc000124a50, 0xc00042b3b0)
	/Users/dgryski/go/src/github.com/quasilyte/go-ruleguard/ruleguard/runner.go:77 +0xab
github.com/quasilyte/go-ruleguard/internal/mvdan.cc/gogrep.(*Pattern).MatchNode(0xc0002f4800, 0x1471160, 0xc000124a50, 0xc0001cef18)
	/Users/dgryski/go/src/github.com/quasilyte/go-ruleguard/internal/mvdan.cc/gogrep/kludge.go:41 +0xb4
github.com/quasilyte/go-ruleguard/ruleguard.(*rulesRunner).run.func2(0x1471160, 0xc000124a50, 0xc00038f901)
	/Users/dgryski/go/src/github.com/quasilyte/go-ruleguard/ruleguard/runner.go:76 +0x15b
go/ast.inspector.Visit(0xc000044890, 0x1471160, 0xc000124a50, 0x146fb80, 0xc000044890)
	/Users/dgryski/go/src/go.googlesource.com/go/src/go/ast/walk.go:373 +0x3a
go/ast.Walk(0x146fb80, 0xc000044890, 0x1471160, 0xc000124a50)
	/Users/dgryski/go/src/go.googlesource.com/go/src/go/ast/walk.go:52 +0x66
go/ast.Walk(0x146fb80, 0xc000044890, 0x14717a0, 0xc0000111c0)
	/Users/dgryski/go/src/go.googlesource.com/go/src/go/ast/walk.go:230 +0x1c44
go/ast.walkStmtList(0x146fb80, 0xc000044890, 0xc00038f9c0, 0x1, 0x1)
	/Users/dgryski/go/src/go.googlesource.com/go/src/go/ast/walk.go:32 +0x9e
go/ast.Walk(0x146fb80, 0xc000044890, 0x14711a0, 0xc000124ae0)
	/Users/dgryski/go/src/go.googlesource.com/go/src/go/ast/walk.go:224 +0x1adf
go/ast.Walk(0x146fb80, 0xc000044890, 0x1471620, 0xc000124b10)
	/Users/dgryski/go/src/go.googlesource.com/go/src/go/ast/walk.go:344 +0xd98
go/ast.walkDeclList(0x146fb80, 0xc000044890, 0xc000010f80, 0x4, 0x4)
	/Users/dgryski/go/src/go.googlesource.com/go/src/go/ast/walk.go:38 +0x9e
go/ast.Walk(0x146fb80, 0xc000044890, 0x14715a0, 0xc000110980)
	/Users/dgryski/go/src/go.googlesource.com/go/src/go/ast/walk.go:353 +0x264e
go/ast.Inspect(...)
	/Users/dgryski/go/src/go.googlesource.com/go/src/go/ast/walk.go:385
github.com/quasilyte/go-ruleguard/ruleguard.(*rulesRunner).run(0xc0000105c0, 0xc000110980, 0x0, 0x1780108)
	/Users/dgryski/go/src/github.com/quasilyte/go-ruleguard/ruleguard/runner.go:72 +0x24f
github.com/quasilyte/go-ruleguard/ruleguard.RunRules(...)
	/Users/dgryski/go/src/github.com/quasilyte/go-ruleguard/ruleguard/ruleguard.go:30
github.com/quasilyte/go-ruleguard/analyzer.runAnalyzer(0xc0000c4180, 0xc0000c4180, 0x0, 0x0, 0x0)
	/Users/dgryski/go/src/github.com/quasilyte/go-ruleguard/analyzer/analyzer.go:70 +0x208
golang.org/x/tools/go/analysis/internal/checker.(*action).execOnce(0xc00010a6e0)
	/Users/dgryski/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:690 +0x83a
sync.(*Once).doSlow(0xc00010a6e0, 0xc00031af90)
	/Users/dgryski/go/src/go.googlesource.com/go/src/sync/once.go:66 +0xec
sync.(*Once).Do(...)
	/Users/dgryski/go/src/go.googlesource.com/go/src/sync/once.go:57
golang.org/x/tools/go/analysis/internal/checker.(*action).exec(0xc00010a6e0)
	/Users/dgryski/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:579 +0x60
golang.org/x/tools/go/analysis/internal/checker.execAll.func1(0xc00010a6e0)
	/Users/dgryski/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:567 +0x34
created by golang.org/x/tools/go/analysis/internal/checker.execAll
	/Users/dgryski/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:573 +0x119
```

--- a/analyzer/testdata/src/filtertest/f1.go
+++ b/analyzer/testdata/src/filtertest/f1.go
@@ -1,5 +1,7 @@
 package filtertest
 
+import "time"
+
 type implementsAll struct{}
 
 func (implementsAll) Read([]byte) (int, error) { return 0, nil }
@@ -46,6 +48,15 @@ func detectType() {
 	typeTest([100]byte{}, "size!=100")
 	typeTest([105]byte{}, "size!=100") // want `YES`
 	typeTest([10]byte{}, "size!=100")  // want `YES`
+
+	var time1, time2 time.Time
+	var err error
+	typeTest(time1 == time2, "time==time") // want `YES`
+	typeTest(err == nil, "time==time")
+	typeTest(nil == err, "time==time")
+	typeTest(time1 != time2, "time!=time") // want `YES`
+	typeTest(err != nil, "time!=time")
+	typeTest(nil != err, "time!=time")
 }
 
 func detectPure(x int) {

--- a/analyzer/testdata/src/filtertest/rules.go
+++ b/analyzer/testdata/src/filtertest/rules.go
@@ -47,6 +47,9 @@ func _(m fluent.Matcher) {
 	m.Match(`typeTest($x, "size==100")`).Where(m["x"].Type.Size == 100).Report(`YES`)
 	m.Match(`typeTest($x, "size!=100")`).Where(m["x"].Type.Size != 100).Report(`YES`)
 
+	m.Match(`typeTest($t0 == $t1, "time==time")`).Where(m["t0"].Type.Is("time.Time")).Report(`YES`)
+	m.Match(`typeTest($t0 != $t1, "time!=time")`).Where(m["t1"].Type.Is("time.Time")).Report(`YES`)
+
 	m.Match(`pureTest($x)`).
 		Where(m["x"].Pure).
 		Report("pure")

--- a/ruleguard/typematch/typematch.go
+++ b/ruleguard/typematch/typematch.go
@@ -322,9 +322,16 @@ func (p *Pattern) matchIdentical(sub *pattern, typ types.Type) bool {
 		if !ok {
 			return false
 		}
+		obj := typ.Obj()
+		pkg := obj.Pkg()
+		// pkg can be nil for builtin named types.
+		// There is no point in checking anything else as we never
+		// generate the opNamed for such types.
+		if pkg == nil {
+			return false
+		}
 		pkgPath := sub.value.([2]string)[0]
 		typeName := sub.value.([2]string)[1]
-		obj := typ.Obj()
 		return obj.Pkg().Path() == pkgPath && typeName == obj.Name()
 
 	default:

